src/components/output/custom-els/MultiPanel/index.ts

```diff
@@ -0,0 +1,208 @@
+      // (= new child was added earlier), skip the ID/tabindex setting activities
+      // and just flag tabindexAlreadySet
+      if (heading.classList.contains('panel-heading')
+        && content.classList.contains('panel-content')) {
+        tabindexAlreadySet = true;
+        continue;
+      }
+
+      // for A11y add id and tab index
+      heading.classList.add('panel-heading');
+      heading.setAttribute('tabindex', '-1');
+      heading.id = `panel-heading-${randomId}`;
+      heading.setAttribute('aria-controls', `panel-content-${randomId}`);
+
+      content.classList.add('panel-content');
+      content.id = `panel-content-${randomId}`;
+      content.setAttribute('aria-labelledby', `panel-heading-${randomId}`);
+    }
+
+    // Only if tab index was never set (= initial load) make the first heading focusable.
+    // otherwise keep tab index where already is assigned.
+    if (!tabindexAlreadySet) {
+      children[0].setAttribute('tabindex', '0');
+    }
+  }
+
+  _getClosestHeading (el: Element) {
+    const closestEl = el.closest('multi-panel > *');
+    if (closestEl && closestEl.classList.contains('panel-heading')) {
+      return closestEl as HTMLElement;
+    }
+    return undefined;
+  }
+
+  // returns headding that is before currently selected one.
+  _prevHeading () {
+    // activeElement would be the currently selected headding
+    // 2 elemements before that would be the previouse heading unless it is the first element.
+    if (this.firstElementChild === document.activeElement) {
+      return this.firstElementChild as HTMLElement;
+    }
+    // previouse Element of active Element is previouse Content,
+    // previouse Element of previouse Content  is previouseHeadding
+    const previouseContent = document.activeElement.previousElementSibling;
+    if (previouseContent) {
+      return previouseContent.previousElementSibling as HTMLElement;
+    }
+    return;
+  }
+
+  // returns headding that is after currently selected one.
+  _nextHeading () {
+    // activeElement would be the currently selected headding
+    // 2 elemements after that would be the next heading.
+    const nextContent = document.activeElement.nextElementSibling;
+    if (nextContent) {
+      return nextContent.nextElementSibling as HTMLElement;
+    }
+    return;
+  }
+
+  // returns first heading in multi-panel.
+  _firstHeading () {
+    // first element is always first headding
+    return this.firstElementChild as HTMLElement;
+  }
+
+  // returns last heading in multi-panel.
+  _lastHeading () {
+    // if the last element is headding, return last element
+    const lastEl = this.lastElementChild as HTMLElement;
+    if (lastEl && lastEl.classList.contains('panel-heading')) {
+      return lastEl;
+    }
+    // otherwise return 2nd from the last
+    const lastContent = this.lastElementChild;
+    if (lastContent) {
+      return lastContent.previousElementSibling as HTMLElement;
```

Comment:
If we type the method return (`private _lastHeading(): HTMLElement {`), are we able to drop the `as` casting here? Or was TypeScript saying `previousElementSibling` was some other type?

Reply:
yeah it will yell at `not assignable`

Reply:
never mind this was resolved! typescript 🤷‍♀️